src/main/java/ca/uhn/fhir/jpa/starter/service/HelperService.java

```diff
@@ -647,45 +648,86 @@ public ResponseEntity<LinkedHashMap<String, Object>> createUsers(@RequestParam("
 
 			if (!(practitioners.contains(firstName) && practitioners.contains(lastName)
 				&& practitioners.contains(phoneNumber + countryCode))) {
-				Practitioner practitioner = FhirResourceTemplateHelper.hcw(firstName, lastName, phoneNumber,
-					countryCode, gender, birthDate, state, lga, ward, facilityUID, role, qualification,
-					stateIdentifier, argusoftIdentifier);
-//				practitionerId = createResource(hcw,
-//					Practitioner.class,
-//					Practitioner.GIVEN.matches().value(hcw.getName().get(0).getGivenAsSingleString()),
-//					Practitioner.FAMILY.matches().value(hcw.getName().get(0).getFamily()),
-//					Practitioner.TELECOM.exactly().systemAndValues(ContactPoint.ContactPointSystem.PHONE.toCode(), Arrays.asList(phoneNumber + countryCode))
-//				); // Catch index out of bound
-				practitioners.add(practitioner.getName().get(0).getFamily());
-				practitioners.add(practitioner.getName().get(0).getGivenAsSingleString());
-				practitioners.add(practitioner.getTelecom().get(0).getValue());
-				PractitionerRole practitionerRole = FhirResourceTemplateHelper.practitionerRole(role, qualification,
-					practitioner.getIdElement().getIdPart(), organizationId);
-//				practitionerRoleId = createResource(practitionerRole, PractitionerRole.class, PractitionerRole.PRACTITIONER.hasId(practitionerId));
-				UserRepresentation user = KeycloakTemplateHelper.user(firstName, lastName, email, keycloakUserName,
-					initialPassword, phoneNumber, countryCode, practitioner.getIdElement().getIdPart(),
-					practitionerRole.getIdElement().getIdPart(), role, state, lga, ward, facilityUID,
-					argusoftIdentifier, countryName);
-				String keycloakUserId = createKeycloakUser(user);
-				if (keycloakUserId == null) {
-					map.put("User not created", s);
+				IBaseResource existingPractitionerResource = fetchExistingFhirResource(Practitioner.class,
+					Practitioner.GIVEN.matchesExactly().value(firstName),
+					Practitioner.FAMILY.matchesExactly().value(lastName),
+					Practitioner.TELECOM.exactly().systemAndValues(ContactPoint.ContactPointSystem.PHONE.toCode(),
+						Arrays.asList(countryCode + phoneNumber)));
+				if (existingPractitionerResource != null ){
+					IBaseResource existingPractitionerRoleResource = fetchExistingFhirResource(PractitionerRole.class,
+						PractitionerRole.PRACTITIONER.hasId("Practitioner/" + existingPractitionerResource.getIdElement().getIdPart()));
```

Comment:
Instead of hardcoding Practitioner/ you can do this as well Practitioner.class.name and concat / to it.